### PR TITLE
modify trace api test=develop

### PR DESCRIPTION
**Pull request by yaoxuefeng6**
### PR types
New features

### PR changes
 APIs

### Describe
 
According paddle 2.0 standard
1, change trace api attr name 'dim' to 'axis'.
2, delete out argument
3, change example code to imperative mode.
4, change api input name 'input' to 'x'.

### CODE REVIEW

**Review by lfchener — DISMISSED** [review]
LGTM

File: python/paddle/tensor/math.py [hunk 1]
```diff
@@ -191,8 +191,8 @@ def func(x, name=None, out=None):
 @templatedoc()
 def pow(input, exponent, out=None, name=None):
     """
-	:alias_main: paddle.pow
-	:alias: paddle.pow,paddle.tensor.pow,paddle.tensor.math.pow
+	:alias_main: paddle.pow
+	:alias: paddle.pow,paddle.tensor.pow,paddle.tensor.math.pow
```
Comment (jzhang533) [inline python/paddle/tensor/math.py:195]: 这里不小心改了行尾，引入了^M，看一下editor的设置。
其他的在PR里看起来删掉了两行和加上的两行一样的地方，也一样。

File: paddle/fluid/operators/trace_op.cc [hunk 2]
```diff
@@ -84,13 +84,13 @@ class TraceOpMaker : public framework::OpProtoAndCheckerMaker {
         )DOC")
         .SetDefault(0);
     AddAttr<int>(
-        "dim1",
+        "axis1",
         R"DOC((int, default 0), the first dim of the 2-D planes from which the diagonals should be taken. 
         Can be both positive and negative. Default: 0.
```
Comment (jzhang533) [inline paddle/fluid/operators/trace_op.cc]: first dim --> first axis
both/and  -> either/or ?

File: paddle/fluid/operators/trace_op.cc [hunk 3]
```diff
@@ -84,13 +84,13 @@ class TraceOpMaker : public framework::OpProtoAndCheckerMaker {
...
         )DOC")
         .SetDefault(-2);
     AddAttr<int>(
-        "dim2",
+        "axis2",
         R"DOC((int, default 1), the second dim of the 2-D planes from which the diagonals should be taken. 
         Can be both positive and negative. Default: 1.
```
Comment (jzhang533) [inline paddle/fluid/operators/trace_op.cc]: second dim --> second axis
both/and  -> either/or ?

File: python/paddle/incubate/complex/tensor/math.py [hunk 4]
```diff
@@ -236,7 +236,7 @@ def elementwise_div(x, y, axis=-1, name=None):
         name=name)
 
 
-def trace(input, offset=0, dim1=0, dim2=1, name=None):
+def trace(input, offset=0, axis1=0, axis2=1, name=None):
```
Comment (jzhang533) [inline python/paddle/incubate/complex/tensor/math.py]: input -> x
according to the latest argument convention.

File: python/paddle/incubate/complex/tensor/math.py [hunk 5]
```diff
@@ -246,8 +246,8 @@ def trace(input, offset=0, dim1=0, dim2=1, name=None):
         input(ComplexVariable): The input ComplexVariable. Must be at least 2-dimensional. 
             The supported data types include complex64 and complex128.
         offset(int, optional): Which diagonals in input tensor will be taken. Default: 0 (main diagonals).
-        dim1(int, optional): The first dimension with respect to take diagonal. Default: 0.
-        dim2(int, optional): The second dimension with respect to take diagonal. Default: 1.
+        axis1(int, optional): The first dimension with respect to take diagonal. Default: 0.
+        axis2(int, optional): The second dimension with respect to take diagonal. Default: 1.
```
Comment (jzhang533) [inline python/paddle/incubate/complex/tensor/math.py]: dimension -> axis

File: python/paddle/incubate/complex/tensor/math.py [hunk 6]
```diff
@@ -264,11 +264,11 @@ def trace(input, offset=0, dim1=0, dim2=1, name=None):
             
             with dg.guard():
                 case1 = dg.to_variable(case1)
-                data1 = paddle.complex.trace(case1, offset=1, dim1=1, dim2=2) # data1.shape = [3]
+                data1 = paddle.complex.trace(case1, offset=1, axis1=1, axis2=2) # data1.shape = [3]
```
Comment (jzhang533) [inline python/paddle/incubate/complex/tensor/math.py]: 2.0 中的示例代码用paddle.enable_imperative()开启动态图。

File: python/paddle/tensor/math.py [hunk 7]
```diff
@@ -1572,17 +1572,17 @@ def clamp(input, min=None, max=None, output=None, name=None):
 
     return output
 
-def trace(input, offset=0, dim1=0, dim2=1, out=None, name=None):
+def trace(input, offset=0, axis1=0, axis2=1, name=None):
```
Comment (jzhang533) [inline python/paddle/tensor/math.py]: input -> x
according to the latest convention.

File: python/paddle/tensor/math.py [hunk 8]
```diff
@@ -1572,17 +1572,17 @@ def clamp(input, min=None, max=None, output=None, name=None):
...
+	:alias_main: paddle.trace
+	:alias: paddle.trace,paddle.tensor.trace,paddle.tensor.math.trace
 
     This OP computes the sum along diagonals of the input tensor.
     
     If ``input`` is 2D, returns the sum of diagonal. 
 
     If ``input`` has larger dimensions, then returns an tensor of diagonals sum, diagonals be taken from
-    the 2D planes specified by dim1 and dim2. By default, the 2D planes formed by the first and second dimensions 
+    the 2D planes specified by axis1 and axis2. By default, the 2D planes formed by the first and second dimensions
```
Comment (jzhang533) [inline python/paddle/tensor/math.py]: first and second dimensions -> first and second axes.

File: python/paddle/tensor/math.py [hunk 9]
```diff
@@ -1594,8 +1594,8 @@ def trace(input, offset=0, dim1=0, dim2=1, out=None, name=None):
     Args:
         input(Variable): The input tensor. Must be at least 2-dimensional. The input data type should be float32, float64, int32, int64.
         offset(int, optional): Which diagonals in input tensor will be taken. Default: 0 (main diagonals).
-        dim1(int, optional): The first dimension with respect to take diagonal. Default: 0.
-        dim2(int, optional): The second dimension with respect to take diagonal. Default: 1.
+        axis1(int, optional): The first dimension with respect to take diagonal. Default: 0.
+        axis2(int, optional): The second dimension with respect to take diagonal. Default: 1.
```
Comment (jzhang533) [inline python/paddle/tensor/math.py]: dimension -> axis

File: python/paddle/tensor/math.py [hunk 10]
```diff
@@ -1617,11 +1617,11 @@ def trace(input, offset=0, dim1=0, dim2=1, out=None, name=None):
                 case2 = dg.to_variable(case2)
                 case3 = dg.to_variable(case3)
                 data1 = paddle.trace(case1) # data1.shape = [1]
-                data2 = paddle.trace(case2, offset=1, dim1=1, dim2=2) # data2.shape = [3]
-                data3 = paddle.trace(case3, offset=-3, dim1=1, dim2=-1) # data2.shape = [3, 5]
+                data2 = paddle.trace(case2, offset=1, axis1=1, axis2=2) # data2.shape = [3]
+                data3 = paddle.trace(case3, offset=-3, axis1=1, axis2=-1) # data2.shape = [3, 5]
     """
```
Comment (jzhang533) [inline python/paddle/tensor/math.py:1622]: use paddle.enable_imperative()

File: python/paddle/tensor/math.py [hunk 11]
```diff
@@ -1634,45 +1634,42 @@ def __check_input(input, offset, dim1, dim2):
...
                 len(input_shape)
 
-        dim1_ = dim1 if dim1 >= 0 else len(input_shape) + dim1
-        dim2_ = dim2 if dim2 >= 0 else len(input_shape) + dim2
+        axis1_ = axis1 if axis1 >= 0 else len(input_shape) + axis1
+        axis2_ = axis2 if axis2 >= 0 else len(input_shape) + axis2
 
-        assert dim1_ < len(input_shape),     \
+        assert axis1_ < len(input_shape),     \
             "The argument dim1 is out of range (expected to be in range of [%d, %d], but got %d).\n"  \
```
Comment (jzhang533) [inline python/paddle/tensor/math.py]: the argument axis1

File: python/paddle/tensor/math.py [hunk 12]
```diff
@@ -1634,45 +1634,42 @@ def __check_input(input, offset, dim1, dim2):
...
+        assert axis2_ < len(input_shape),   \
+            "The argument axis2 is out of range (expected to be in range of [%d, %d], but got %d).\n"   \
+            % (-(len(input_shape)), len(input_shape) - 1, axis2)
 
 
-        assert  dim1_ != dim2_,   \
-               "dim1 and dim2 cannot be the same dimension." \
-                "But received dim1 = %d, dim2 = %d\n"%(dim1, dim2)
+        assert  axis1_ != axis2_,   \
+               "axis1 and axis2 cannot be the same dimension." \
```
Comment (jzhang533) [inline python/paddle/tensor/math.py]: dimension -> axis

**Review by jzhang533 — COMMENTED** [review]
see inline comments

File: python/paddle/tensor/math.py (hunk 1 above)
Comment (jzhang533) [reply python/paddle/tensor/math.py:195]: 好像是原来的某个commit引入的^M。

**Review by jzhang533 — APPROVED** [review]
LGTM

**Review by liym27 — APPROVED** [review]
LGTM for the changes of Attrs of trace op, but please node that it will cause that Paddle inference library of 2.0 version to load model trained by the old version failed.

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**